caas/kubernetes/provider/k8s.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package provider
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"gopkg.in/juju/names.v2"
+	"k8s.io/client-go/kubernetes"
+	k8serrors "k8s.io/client-go/pkg/api/errors"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/juju/juju/agent"
+	"github.com/juju/juju/caas"
+	"github.com/juju/juju/environs"
+)
+
+var logger = loggo.GetLogger("juju.kubernetes.provider")
+
+// TODO(caas) should be using a juju specific namespace
+const namespace = "default"
```

Comment:
Can we make this user configurable? not in this PR, I just wonder if it really needs to be juju-specific. Can't we use object annotations to track what's managed by Juju?

Reply:
yes, the intent is to make it configurable.
i will be going through the TODO(caas) items and creating cards